airflow/contrib/hooks/sagemaker_hook.py

```diff
@@ -0,0 +1,811 @@
+        None implies no timeout for any SageMaker job.
+        :type max_ingestion_time: int
+        :return: A response to endpoint update
+        """
+
+        response = self.conn.update_endpoint(
+            **config)
+        if wait_for_completion:
+            self.check_status(config['EndpointName'],
+                              SageMakerHook.non_terminal_states,
+                              SageMakerHook.failed_states,
+                              'EndpointStatus',
+                              self.describe_endpoint,
+                              check_interval, max_ingestion_time
+                              )
+        return response
+
+    def describe_training_job(self, name):
+        """
+        :param name: the name of the training job
+        :type name: str
+        Return the training job info associated with the current job_name
+        :return: A dict contains all the training job info
+        """
+
+        return self.conn\
+                   .describe_training_job(TrainingJobName=name)
+
+    def describe_training_job_with_log(self, job_name, non_terminal_states, positions, stream_names, instance_count,
+                                       state, last_description, last_describe_job_call):
+        log_group = '/aws/sagemaker/TrainingJobs'
+
+        if len(stream_names) < instance_count:
+            # Log streams are created whenever a container starts writing to stdout/err, so this list
+            # may be dynamic until we have a stream for every instance.
+            try:
+                streams = self.log_conn.describe_log_streams(logGroupName=log_group, logStreamNamePrefix=job_name + '/',
+                                                             orderBy='LogStreamName', limit=instance_count)
+                stream_names = [s['logStreamName'] for s in streams['logStreams']]
+                positions.update([(s, Position(timestamp=0, skip=0))
+                                  for s in stream_names if s not in positions])
+            except ClientError as e:
+                # On the very first training job run on an account, there's no log group until
+                # the container starts logging, so ignore any errors thrown about that
+                err = e.response.get('Error', {})
+                if err.get('Code', None) != 'ResourceNotFoundException':
+                    raise
+
+        if len(stream_names) > 0:
+            for idx, event in self.multi_stream_iter(log_group, stream_names, positions):
+                self.log.info(event['message'])
+                ts, count = positions[stream_names[idx]]
+                if event['timestamp'] == ts:
+                    positions[stream_names[idx]] = Position(timestamp=ts, skip=count + 1)
+                else:
+                    positions[stream_names[idx]] = Position(timestamp=event['timestamp'], skip=1)
+        else:
+            sys.stdout.flush()
```

Comment:
What's this for? I don't see us printing to stdout directly anywhere? Have I missed it?